### PR TITLE
[chore] prepare v0.108.0 release

### LINKED ISSUE #3241: Release the operator v0.108.0

### PR DESCRIPTION
**Pull request by TylerHelmuth**
closes #3241

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -759,6 +759,7 @@ The OpenTelemetry Operator _might_ work on versions outside of the given range,
 
 | OpenTelemetry Operator | Kubernetes     | Cert-Manager | Prometheus-Operator |
 |------------------------|----------------| ------------ |---------------------|
+| v0.108.0               | v1.23 to v1.30 | v1           | v0.75.0             |
```
Comment (jaronoff97) [inline README.md:762]: we support up to 1.31 now https://github.com/open-telemetry/opentelemetry-operator/pull/3246

File: versions.txt [hunk 2]
```diff
@@ -2,16 +2,16 @@
 # by default with the OpenTelemetry Operator. This would usually be the latest
 # stable OpenTelemetry version. When you update this file, make sure to update the
 # the docs as well.
-opentelemetry-collector=0.107.0
+opentelemetry-collector=0.108.0
```
[the remark this answers is not in the document]
Comment (jaronoff97) [reply versions.txt:5]: nah let's not mess with those unless we really need to (i.e. it's something we had added automagically previously)

File: README.md (hunk 1 above)
Comment (TylerHelmuth) [reply README.md]: Updated

**Review by jaronoff97 — APPROVED** [review]

File: CHANGELOG.md [hunk 3]
```diff
@@ -2,6 +2,47 @@
...
+- `collector`: Allow autoscaler targetCPUUtilization and TargetMemoryUtilization to be greater than 99 (#3258)
+- `auto-instrumentation`: Not ignore the `instrumentation.opentelemetry.io/container-names` annotation when the multi-instrumentation is enabled (#3090)
+- `operator`: Support for Kubernetes `1.31` version. (#3247)
+- `target allocator`: introduces the global field in the TA config to allow for setting scrape protocols (#3160)
+
+### 🧰 Bug fixes 🧰
+
+- `auto-instrumentation`: Fix file copy for NGINX auto-instrumentation for non-root workloads. (#2726)
+- `target allocator`: Retrying failed namespace informer creation in promOperator CRD watcher, then exit if creation issue cannot be resolved (#3216)
+- `target allocator`: Rollback 3187 (#3242)
```
Comment (swiatekm) [inline CHANGELOG.md:19]:
```suggestion
- `target allocator`: Rollback #3187 (#3242)
```

Let's make this a bit more readable.

File: README.md [hunk 4]
```diff
@@ -759,6 +759,7 @@ The OpenTelemetry Operator _might_ work on versions outside of the given range,
 
 | OpenTelemetry Operator | Kubernetes     | Cert-Manager | Prometheus-Operator |
 |------------------------|----------------| ------------ |---------------------|
+| v0.108.0               | v1.23 to v1.31 | v1           | v0.75.0             |
```
Comment (swiatekm) [inline README.md:762]:
```suggestion
| v0.108.0               | v1.23 to v1.31 | v1           | v0.76.0             |
```

**Review by swiatekm — APPROVED** [review]

**Pull request merged**